loopy/transform/iname.py

```diff
@@ -1126,26 +1128,56 @@ def has_schedulable_iname_nesting(kernel):
 # {{{ rename_inames
 
 @for_each_kernel
-def rename_iname(kernel, old_iname, new_iname, existing_ok=False, within=None):
+def rename_inames(kernel, old_inames, new_iname, existing_ok=False, within=None):
     """
+    :arg old_inames: A collection of inames that must be renamed to **new_iname**.
     :arg within: a stack match as understood by
         :func:`loopy.match.parse_stack_match`.
     :arg existing_ok: execute even if *new_iname* already exists
     """
+    from collections.abc import Collection
```

Comment:
Maybe check that `new_iname` doesn't occur in `old_inames`?

Reply:
Done, thanks! We raise an error for this now.